packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -1353,27 +1360,33 @@ export default class InternalModel {
     this.store.recordArrayManager.recordDidChange(this.identifier);
   }
 
-  setId(id: string) {
+  setId(id: string, fromCache: boolean = false) {
+    if (this._isUpdatingId === true) {
```

Comment:
`if (this._isUpdatingId)`, no?

Reply:
I prefer strict checks myself but yes, we could shorthand.